v2/http-middleware/security.go

```diff
@@ -201,65 +199,87 @@ type Authorizer interface {
 func AuthorizeMiddleware(authorizer Authorizer) mux.MiddlewareFunc {
 	return func(next http.Handler) http.Handler {
 		return http.HandlerFunc(func(w http.ResponseWriter, req *http.Request) {
-			ctx, span := trace.StartSpan(req.Context(), "Authorizer")
+			ctx, span := trace.StartSpan(req.Context(), "AuthorizeMiddleware/Handler")
 			defer span.End()
 
-			logFields := log.
-				WithTracing(ctx).
-				WithField("method", req.Method).
-				WithField("url", req.URL.String())
-
-			secConfig := lookupSecurityConfig(req)
-			if len(secConfig.authorizations) == 0 {
-				span.End()
-				next.ServeHTTP(w, req)
-				return
-			}
-
 			userID, ok := useridcontext.FromContext(req.Context())
 			if !ok {
-				logFields.Error("Couldn't extract User ID from context.")
+				log.WithTracing(ctx).
+					WithField("method", req.Method).
+					WithField("url", req.URL.String()).
+					Error("Couldn't extract User ID from context")
+
 				http_server.WriteJSONResponse(
 					ctx, w, req, http.StatusInternalServerError, http_model.ErrResponseInternalServerError,
 				)
 				return
 			}
 
-			logFields = logFields.WithUserID(ctx)
-
-			for _, authorizeConfig := range secConfig.authorizations {
-				resource, err := authorizeConfig.resourceFunc(req)
-				if err != nil {
-					logFields.WithError(err).Error("ResourceFunc failed.")
-					http_server.WriteJSONResponse(
-						ctx, w, req, http.StatusInternalServerError, http_model.ErrResponseInternalServerError,
-					)
-					return
-				}
+			isAuthorized, err := checkAuthorization(ctx, req, authorizer, userID)
+			if err != nil {
+				http_server.WriteJSONResponse(ctx, w, req, http.StatusInternalServerError, http_model.ErrResponseInternalServerError)
+				return
+			}
 
-				ok, err := authorizer.IsAuthorizedWithContext(
-					ctx,
-					userID,
-					authorizeConfig.action,
-					resource,
-				)
-				if !ok || err != nil {
-					logFields.
-						WithField("userId", userID).
-						WithField("action", authorizeConfig.action).
-						WithField("resource", resource).
-						Warn("User is not Authorized")
-					http_server.WriteJSONResponse(ctx, w, req, http.StatusUnauthorized, http_model.ErrResponseUnauthorized)
-					return
-				}
+			if !isAuthorized {
+				http_server.WriteJSONResponse(ctx, w, req, http.StatusUnauthorized, http_model.ErrResponseUnauthorized)
+				return
 			}
 
-			span.End()
 			next.ServeHTTP(w, req)
```

Comment:
we had the `span.End()` before `next.ServeHTTP(w, req)` so that the api-handler won't be a sub-span to the authorize-handler

Reply:
just realised, that it's no longer needed, so we should remove the other `span.End()` as well